notebooks/10_bayesian_networks_sampling/index.md

```diff
@@ -0,0 +1,233 @@
+
+# Prior Sampling
+Now we're ready to tackle the sampling problem! Suppose we want to sample from the joint distribution 
+of the Bayes' Net, $Pr(x_1,...,x_n)$.
+
+Consider the following process:
+
+1.&nbsp; For i = 1 to n
+
+2.&emsp;&emsp; Sample $x_i$ from $Pr(x_i | parents(x_i))$
+
+3.&nbsp; Return $(x_1,...,x_n)$
+
+This process generates a sample $X=(x_1,...,x_n)$ with the following probability:
+$$
+S_{PS}(x_1,...,x_n) = \prod_{i=1}^{n}Pr(x_i|parents(x_i)) = Pr(x_1,...,x_n),
+$$
+which is the Bayes' Net's joint probability.
+
+Now we need to prove that the samples are realizations of I.I.D (Independent, Identically Distributed) random variables. 
+1. Since we choose a new random number for every sample, thus samples are independent from each other. 
+2. It has been also shown that they are identically distributed because of the formula above.
+
+Suppose we have drawn $N$ samples from the joint distribution. Let the number of samples of an event be 
+$N_{PS}(x_1,...,x_n)$.
+
+So based on the law of large numbers, we have:
+$$
+lim_{N\to\infty}\hat{Pr}(x_1,...,x_n) = lim_{N\to\infty} \frac{N_{PS}(x_1,...,x_n)}{N}
+$$
+$$
+= S_{PS}(x_1,...,x_n) = Pr(x_1,...,x_n),
+$$
+
+hence, the sampling procedure is consistent with the joint distribution.
+
+It is apparent that this algorithm is faster than its exact counter-parts. Since we have the joint distribution, we can calculate the probability of any event. However, in the case of conditional probabilities, it's more efficient not to consider samples inconsistent with the evidence. This brings us to the idea of rejection sampling.
+
+# Rejection Sampling
+
+One minor problem with Prior Sampling is that we keep samples which are not consistent with evidence and since the evidence might be unlikely, the number of unused samples (samples which will be discarded due to inconsistency with the evidence) will eventually be great.
+A simple idea is to reject incosistent samples whilst samling, to achieve this goal, whenever an incosistent sample observed, we will ignore (reject) that sample.
+
+Consider the following process:
+
+1.&nbsp; For i = 1 to n
+
+2.&emsp;&emsp; Sample $x_i$ from $Pr(x_i | parents(x_i))$
+
+3.&emsp;&emsp; if $x_i$ not consistent with evidence:
+
+4.&emsp;&emsp;&emsp;return (no sample is generated in this cycle)
+
+5.&nbsp; Return $(x_1,...,x_n)$
+
+It is also consistent for conditional probabilities.
+
+# Likelihood Weighting
+
+If we look closer to the problem with Prior Sampling, which lead us to Rejection Sampling method, we see that if the evidence is unlikely, many samples will be rejected, thus we end up repeating sampling process many times to achieve the desired sample size. This problem brings us to Likelihood Weighting. The idea is to fix the evidence variables and sample the rest, But it will cause inconsistency with the distribution. The solution is to use a weight variable indicating the probability of evidences given their parents.
```

Comment:
lead -> led

Reply:
Thank you for your notice. It has been resolved.